### PR TITLE
fix: adjust sequence number for app-id compatibility with palo alto fw

### PR DESCRIPTION
**Pull request by sibicramesh**
https://github.com/aporeto-inc/enforcerd/pull/1948

### CODE REVIEW

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "c7c5ab8b07bcd0277445091f3024be69e8a2e5f6",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "57da42e12150eebe916669336380b75c62904a6a",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "57da42e12150eebe916669336380b75c62904a6a",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "ebaf65a9e97f2e5aff1d1ddcaef4198324fb47b8",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "57da42e12150eebe916669336380b75c62904a6a",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "e21fc17e8357cc8c2797e82f657c6870ed4e40e8",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "57da42e12150eebe916669336380b75c62904a6a",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "fe231c5edba615c077956bab825c2bfcb4004868",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "57da42e12150eebe916669336380b75c62904a6a",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "11022ad9fa76a5a9f441a6c9287efb6f0ef1b850",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "57da42e12150eebe916669336380b75c62904a6a",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "ed544c31376b6d169718a1692566b76ff9a55f5b",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "57da42e12150eebe916669336380b75c62904a6a",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "ed544c31376b6d169718a1692566b76ff9a55f5b",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "f689cbb4589527b62d98e790f4270c6208f86db7",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "ed544c31376b6d169718a1692566b76ff9a55f5b",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "f689cbb4589527b62d98e790f4270c6208f86db7",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "55d779ba957df279072bcce55328191b7019f767",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "997130cc65d43bda937605d0023f37a9d64c41ac",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "e9a7bd31e74ab2a384be73ae9e5819097d59678d",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "191818e56c99cfb1ca280d43f6f3cfa0fb011edd",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "e9a7bd31e74ab2a384be73ae9e5819097d59678d",
    "pipeline": "release-3.14.0"
  }
]
```

**Review by mheese — APPROVED** [review]
Looks clean. @amitlimaye is the authority though about the details of the datapath changes. It is being cleanly applied though only when the flag is set.

File: controller/internal/enforcer/nfqdatapath/datapath.go [hunk 1]
```diff
@@ -48,6 +49,8 @@ var collectCounterInterval = 30 * time.Second
 // GetUDPRawSocket is placeholder for createSocket function. It is useful to mock tcp unit tests.
 var GetUDPRawSocket = afinetrawsocket.CreateSocket
 
+var ackSize uint32 = 136 + uint32(len(tokens.AckPattern))
```
Comment (amitlimaye) [inline controller/internal/enforcer/nfqdatapath/datapath.go]: if 136 is a magic number we introduced there should be a comment saying what does this signify

**Comment by sibicramesh** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "seq-num",
    "component": "enforcerd",
    "pr-id": "1969",
    "commit-sha": "926e87e201562555cb215f8f3df371c2a525f600",
    "pipeline": "release-3.14.0"
  },
  {
    "project": "seq-num",
    "component": "trireme-lib",
    "pr-id": "1035",
    "commit-sha": "8fd66b132c1dfcb5b3359b59ac8b14b250d7b555",
    "pipeline": "release-3.14.0"
  }
]
```

**Comment by mheese** [comment]
confirmed by @Rohit4793 @sibicramesh and @amitlimaye ... merging this now so that we can adjust the Gopkg.lock on the enforcer PR.

**Pull request merged**